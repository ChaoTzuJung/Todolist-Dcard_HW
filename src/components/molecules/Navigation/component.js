import React from 'react';
import classnames from 'classnames';

import styles from './index.css';

const Navigation = ({ className, todoCategorys, activeCategory, onClickTab }) => (
	<div className={classnames(styles.navigation, className)}>
		{todoCategorys.map(category => (
			<div
				key={category.key}
				className={classnames(styles.item, {
					[styles.active]: category.key === activeCategory,
				})}
				role="button"
				tabIndex="0"
				onClick={() => onClickTab(category.key)}
				onKeyPress={() => {}}
			>
				{category.label}
				<div className={styles.bottomLine} />
			</div>
		))}
	</div>
);

export default Navigation;
