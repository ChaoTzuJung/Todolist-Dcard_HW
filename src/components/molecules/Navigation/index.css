.navigation {
	display: flex;

	& > .item {
		display: flex;
		flex-shrink: 0;
		flex-direction: column;
		align-items: center;
		padding: 24px 30px 0 30px;
		color: #00408B;
		font-size: 24px;
		font-weight: 400;
		text-align: center;
		cursor: pointer;

		& > .bottom-line {
			margin-top: 20px;
			width: 160px;
			height: 5px;
			background-color: #00408B;
			opacity: 0;
		}

		&.active,
		&:hover,
		&:active {
			color: #FFFFFF;
			font-weight: 500;

			& > .bottom-line {
				opacity: 1;
			}
		}
	}

}
