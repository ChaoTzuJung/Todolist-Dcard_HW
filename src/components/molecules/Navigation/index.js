import { contain } from 'react-container-helper';

import Navigation from './component';

const initState = ({ activeCategory }) => ({
	activeCategory,
});

const mapSetStateToProps = (
	{ activeCategory },
	{ className, todoCategorys, onClickTab },
	setState,
) => ({
	// state
	activeCategory,

	// props
	className,
	todoCategorys,

	onClickTab(key) {
		setState({ activeCategory: key });
		onClickTab(key);
	},
});

export default contain(initState, mapSetStateToProps)(Navigation);
