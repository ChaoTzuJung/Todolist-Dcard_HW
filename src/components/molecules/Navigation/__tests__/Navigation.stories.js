import React from 'react';
import { withKnobs, object } from '@storybook/addon-knobs/react';

import Navigation from 'components/molecules/Navigation';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

const stories = storiesOf('Commons|molecules/Navigation', module);

stories.addDecorator(withKnobs);

const todoCategorys = [
	{
		id: 0,
		key: 'myTasks',
		label: 'My Tasks',
	},
	{
		id: 1,
		key: 'inProgress',
		label: 'In Progress',
	},
	{
		id: 2,
		key: 'completed',
		label: 'Completed',
	},
];

stories.add('__interactive', () => (
	<Navigation
		className=""
		activeCategory="myTasks"
		onClickTab={action('onClick tab')}
		todoCategorys={object('todoCategorys', todoCategorys)}
	/>
));
