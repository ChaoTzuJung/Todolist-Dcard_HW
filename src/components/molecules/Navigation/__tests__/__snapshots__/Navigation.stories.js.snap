// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<Contain(Navigation)
  activeCategory="myTasks"
  className=""
  onClickTab={[Function]}
  todoCategorys={
    Array [
      Object {
        "id": 0,
        "key": "myTasks",
        "label": "My Tasks",
      },
      Object {
        "id": 1,
        "key": "inProgress",
        "label": "In Progress",
      },
      Object {
        "id": 2,
        "key": "completed",
        "label": "Completed",
      },
    ]
  }
>
  <Navigation
    activeCategory="myTasks"
    className=""
    onClickTab={[Function]}
    todoCategorys={
      Array [
        Object {
          "id": 0,
          "key": "myTasks",
          "label": "My Tasks",
        },
        Object {
          "id": 1,
          "key": "inProgress",
          "label": "In Progress",
        },
        Object {
          "id": 2,
          "key": "completed",
          "label": "Completed",
        },
      ]
    }
  >
    <div
      className="navigation"
    >
      <div
        className="item active"
        key="myTasks"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        My Tasks
        <div
          className="bottomLine"
        />
      </div>
      <div
        className="item"
        key="inProgress"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        In Progress
        <div
          className="bottomLine"
        />
      </div>
      <div
        className="item"
        key="completed"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        Completed
        <div
          className="bottomLine"
        />
      </div>
    </div>
  </Navigation>
</Contain(Navigation)>
`;
